addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/handler/ZoneMinderThingMonitorHandler.java

```diff
@@ -624,265 +739,451 @@ public void updateStatus(ThingStatus status) {
+                    logger.error(
+                            "{}: context='fetchMonitorDaemonStatus' error in call to 'getAnalysisDaemonStatus' - Exception='{}' ",
+                            getLogIdentifier(), zme.getClass().getCanonicalName(), zme.getCause());
+                } catch (Exception ex) {
+                    logger.error(
+                            "{}: context='fetchMonitorDaemonStatus' tag='exception' error in call to 'getAnalysisDaemonStatus' - Exception='{}'",
+                            getLogIdentifier(), ex.getClass().getCanonicalName(), ex);
+                } finally {
+                    dataConverter.setMonitorAnalysisDaemonStatus(stateAnalysis);
 
+                }
+            }
+
+            if (isLinked(ZoneMinderConstants.CHANNEL_MONITOR_FRAME_DAEMON_STATE)) {
+                try {
+                    stateFrame = UnDefType.UNDEF;
+                    if ((fetchFrame) && frameDaemonActive) {
+                        frameDaemon = monitorProxy.getFrameDaemonStatus();
+                        logger.debug(
+                                "{}: context='fetchMonitorDaemonStatus' tag='frameDaemon' URL='{}' ResponseCode='{}' ResponseMessage='{}'",
+                                getLogIdentifier(), frameDaemon.getHttpRequestUrl(), frameDaemon.getHttpStatus(),
+                                frameDaemon.getHttpResponseMessage());
+
+                        if (frameDaemon != null) {
+                            stateFrame = ((frameDaemon.getStatus() && analysisDaemon.getStatus()) ? OnOffType.ON
+                                    : OnOffType.OFF);
+                        }
+                    }
+                } catch (ZoneMinderResponseException zmre) {
+                    logger.error(
+                            "{}: context='fetchMonitorDaemonStatus' error in call to 'getFrameDaemonStatus' - Http: Status='{}', Message='{}', ExceptionMessage'{}', Exception='{}'",
+                            getLogIdentifier(), zmre.getHttpStatus(), zmre.getHttpMessage(), zmre.getExceptionMessage(),
+                            zmre.getClass().getCanonicalName(), zmre.getCause());
+                } catch (ZoneMinderInvalidData zmid) {
+                    logger.error(
+                            "{}: context='fetchMonitorDaemonStatus' error in call to 'getFrameDaemonStatus' - Response='{}', Exception='{}'",
+                            getLogIdentifier(), zmid.getResponseString(), zmid.getClass().getCanonicalName(),
+                            zmid.getCause());
+
+                } catch (ZoneMinderGeneralException | ZoneMinderAuthenticationException zme) {
+                    logger.error(
+                            "{}: context='fetchMonitorDaemonStatus' error in call to 'getFrameDaemonStatus' - Exception='{}'",
+                            getLogIdentifier(), zme.getClass().getCanonicalName(), zme.getCause());
+                } catch (Exception ex) {
```

Comment:
Please catch specific exceptions instead of catching all exceptions. That way it is more clear what exceptions need to be handled and how this is done.